python/cuda_parallel/cuda/parallel/experimental/iterators/_iterators.py

```diff
@@ -142,13 +181,9 @@ def advance(state, distance):
     def dereference(state):
         return state[0][0]
 
-    @property
-    def state(self) -> ctypes.c_void_p:
-        return ctypes.cast(ctypes.pointer(self._cvalue), ctypes.c_void_p)
-
 
-def pointer(container, ntype: types.Type) -> RawPointer:
-    return RawPointer(container.__cuda_array_interface__["data"][0], ntype)
+def pointer(container, value_type: types.Type) -> RawPointer:
```

Comment:
Note: These machinery can also be replaced by StridedMemoryView once cuda.core becomes a dependency.